sdk/core/Azure.Core/src/MultipartFormDataContent.cs

```diff
@@ -0,0 +1,397 @@
+        ///  Add HTTP content to a collection of RequestContent objects that
+        ///  get serialized to multipart/form-data MIME type.
+        /// </summary>
+        /// <param name="content">The Request content to add to the collection.</param>
+        public void Add(RequestContent content)
+        {
+            Argument.AssertNotNull(content, nameof(content));
+            AddInternal(content, null, null, null);
+        }
+
+        /// <summary>
+        ///  Add HTTP content to a collection of RequestContent objects that
+        ///  get serialized to multipart/form-data MIME type.
+        /// </summary>
+        /// <param name="content">The Request content to add to the collection.</param>
+        /// <param name="headers">The headers to add to the collection.</param>
+        public void Add(RequestContent content, Dictionary<string, string> headers)
+        {
+            Argument.AssertNotNull(content, nameof(content));
+            Argument.AssertNotNull(headers, nameof(headers));
+
+            AddInternal(content, headers, null, null);
+        }
+
+        /// <summary>
+        ///  Add HTTP content to a collection of RequestContent objects that
+        ///  get serialized to multipart/form-data MIME type.
+        /// </summary>
+        /// <param name="content">The Request content to add to the collection.</param>
+        /// <param name="name">The name for the request content to add.</param>
+        /// <param name="headers">The headers to add to the collection.</param>
+        public void Add(RequestContent content, string name, Dictionary<string, string>? headers)
+        {
+            Argument.AssertNotNull(content, nameof(content));
+            Argument.AssertNotNullOrWhiteSpace(name, nameof(name));
+
+            AddInternal(content, headers, name, null);
+        }
+
+        /// <summary>
+        ///  Add HTTP content to a collection of RequestContent objects that
+        ///  get serialized to multipart/form-data MIME type.
+        /// </summary>
+        /// <param name="content">The Request content to add to the collection.</param>
+        /// <param name="name">The name for the request content to add.</param>
+        /// <param name="fileName">The file name for the reuest content to add to the collection.</param>
+        /// <param name="headers">The headers to add to the collection.</param>
+        public void Add(RequestContent content, string name, string fileName, Dictionary<string, string>? headers)
+        {
+            Argument.AssertNotNull(content, nameof(content));
+            Argument.AssertNotNullOrWhiteSpace(name, nameof(name));
+            Argument.AssertNotNullOrWhiteSpace(fileName, nameof(fileName));
+
+            AddInternal(content, headers, name, fileName);
+        }
+
+        private void AddInternal(RequestContent content, Dictionary<string, string>? headers, string? name, string? fileName)
+        {
+            if (headers == null || headers.Count == 0)
```

Comment:
I don't think you need `headers.Count == 0` here

Reply:
Yeah, right!